drivers/source/serial_io.c

```diff
@@ -24,29 +23,43 @@ void sciMutexInit(void) {
     }
     /* USER CODE BEGIN */
     // Create mutex to protect SCI2/SCILin module here.
+    if(sci2Mutex == NULL) {
+        sci2Mutex = xSemaphoreCreateBinary();
```

Comment:
I'd suggest using `xSemaphoreCreateMutex` here for a few reasons:
- To stay consistent with `sciMutex`
- As per the [FreeRTOS docs](https://www.freertos.org/xSemaphoreCreateBinary.html#:~:text=Binary%20semaphores%20and,simple%20mutual%20exclusion.), mutexes are usually preferred over binary semaphores when dealing with simple mutual exclusion.